docs/libvirt-pod-networking.md

```diff
@@ -0,0 +1,131 @@
+
+
+## API
+
+The API is pretty clear: As usual an `interface` element will be defined.
+To indicate that pod interfaces shall be used for networking, a new network
+type (`pod-network`) is introduced.
+All other fields, concerning the properties of the interface, can be used as
+usual.
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+      - type: pod-network
+        mac:
+          address: 00:11:22:33:44:55
+        model:
+          type: virtio
+```
+
+
+## Implementation
+
+Prerequisites:
+
+- libvirt is using pod networking (instead of currently using host networking)
+- A CNI proxy to allow pods to perform CNI actions in the context of the host
+  (this needs to be implemented, as part of this feature, or as an dependent
+  one)
+
+Conceptually the implementation works by assuming that every vNIC will be
+mapped to a _new_ pod interface. To achieve this, for every vNIC a _new_ pod
+interface is requested through CNI.
+Another crucial constraint for this design is that in Kubernetes IP addresses
+are getting assigned to pods, not interfaces. Thus this solution needs to
+achieve the same, and provide IP addresses to VMs, not just a interface.
+
+During this document we call each of these _new_ pod interfaces _VM interface_,
+in order to differentiate them from the the originally-allocated pod interface
+(`eth0`). The original pod interface (`eth0`) is never modified by Kubevirt,
+and can be used to access libvirtd (through the libvirtd pod IP) or to provide
+VM-centric services through the VM pod IP.
+
+**Note:** The original pod interface (eth0) is never touched and can be used to
+access libvirtd (through the libvirtd pod IP) or to provide VM centric services
+through the VM pod IP.
+
+The required steps to enable networking as described are:
+
+1. Request/Create a _new_ VM interface on the libvirt pod, by using the CNI
+   proxy to use the hosts CNI configuration and plugins.
+2. Remember the IP of the VM interface, then remove the IP from the VM
+   interface
+3. Create a libvirt network, backed by a bridge, enable DHCP,
+   and configure libvirt to provide the remembered IP address to the VM.
+   Achieved by adding a DHCP host entry, which maps the vNIC MAC address to the
+   remembered IP
+4. On VM shutdown, delete all VM interfaces. They can be infered by following
+   the libvirt networks associated with the vNICs
+
+**Note:** The _new_ VM interface can not be directly attached to the VM,
+because we lose the ability to provide the IP to the guest. The IP can only be
+provided by DHCP, which requires the use of a bridge.
+
+This process must be repeated for every virtual NIC of every VM.
+Thus N virtual NICs correspond to N VM interfaces, and in turn to N IP
+addresses.
+
+**Note:** This might look like an overhead, creating a single VM interface
+per virtual NIC. The reason for doing so is, that this is a Kubernetes/CNI
+compatible way to request IP endpoints from the networking sub-system.
```

Comment:
English:

Creating one VM interface per virtual NIC might look like an overhead, but this is a Kubernetes/CNI-compatible way to request IP endpoints from the networking sub-system.